server/src-lib/Hasura/Eventing/EventTrigger.hs

```diff
@@ -189,6 +189,11 @@ processEventQueue logger logenv httpMgr pool getSchemaCache eeCtx@EventEngineCtx
         let newLockedEvents = Set.union lockedEvents (Set.fromList evtsIds)
         writeTVar _eeCtxLockedEvents newLockedEvents
 
+    removeEventFromLockedEvents :: EventId -> IO ()
+    removeEventFromLockedEvents eventId = do
+      liftIO $ atomically $ do
+        lockedEvents <- readTVar _eeCtxLockedEvents
+        writeTVar _eeCtxLockedEvents $ Set.delete eventId lockedEvents
```

Comment:
It's probably fine, but can you add a strictness annotation here?:

```
writeTVar _eeCtxLockedEvents $! Set.delete eventId lockedEvents
                                                    ^
```
I'm always a little worried about writing thunks to mutable variables

Reply:
Oh yes..I didn't realize that I'm writing a thunk to a mutable variable. I will also change it where we are saving the locked events.